### PR TITLE
Uses custom plugin to publish zips to maven

### PR DESCRIPTION
**Pull request by lezzago**
Signed-off-by: Ashish Agrawal <ashisagr@amazon.com>

*Issue #, if available:*
#435
*Description of changes:*
Uses custom plugin to publish zips to maven
*CheckList:*
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/build.gradle [hunk 1]
```diff
@@ -29,6 +30,17 @@ opensearchplugin {
     extendedPlugins = ['lang-painless']
 }
 
+publishing {
+    publications {
+        pluginZip(MavenPublication) { publication ->
+            pom {
+                name = "opensearch-alerting"
+                description = "OpenSearch Alerting plugin"
+            }
```
Comment (prudhvigodithi) [inline alerting/build.gradle:39]: If not please also add `licenses` and `developers` info.

File: settings.gradle [hunk 2]
```diff
@@ -8,3 +8,4 @@ include 'alerting'
 include 'core'
 
 project(":core").name = 'alerting-core'
+startParameter.excludedTaskNames=["validatePluginZipPom"]
```
Comment (prudhvigodithi) [inline settings.gradle:11]: Since you are configuring the POM above you dont need to exclude this `validatePluginZipPom`.

File: scripts/build.sh [hunk 3]
```diff
@@ -73,3 +73,5 @@ distributions="$(dirname "${zipPath}")"
 
 echo "COPY ${distributions}/*.zip"
 cp ${distributions}/*.zip ./$OUTPUT/plugins
+
+./gradlew publishPluginZipPublicationToZipStagingRepository -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER
```
Comment (prudhvigodithi) [inline scripts/build.sh:77]: Hey once you call this task `publishPluginZipPublicationToZipStagingRepository` make sure you run `cp -r ./build/local-staging-repo/org/opensearch/. $OUTPUT/maven/org/opensearch` 
Example : https://github.com/opensearch-project/job-scheduler/pull/182/files

File: settings.gradle (hunk 2 above)
Comment (lezzago) [reply settings.gradle]: Removed

File: alerting/build.gradle (hunk 1 above)
Comment (lezzago) [reply alerting/build.gradle:51]: Added

File: alerting/build.gradle [hunk 4]
```diff
@@ -29,6 +30,29 @@ opensearchplugin {
...
+                licenses {
+                    license {
+                        name = "The Apache License, Version 2.0"
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
+                    }
+                }
+                developers {
+                    developer {
+                        name = "OpenSearch"
+                        url = "https://github.com/opensearch-project/ml-commons"
```
Comment (qreshi) [inline alerting/build.gradle:48]: Should this be `OpenSearch` link instead of `ml-commons`?

Comment (prudhvigodithi) [reply alerting/build.gradle:48]: good catch @qreshi it should be the pluginName.

**Review by qreshi — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Review by prudhvigodithi — APPROVED** [review]

**Pull request merged**